src/Featurizers/ForecastingPivotFeaturizer.h

```diff
@@ -0,0 +1,129 @@
+// ----------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License
+// ----------------------------------------------------------------------
+#pragma once
+
+#include "../Archive.h"
+#include "../Traits.h"
+
+#include "Components/InferenceOnlyFeaturizerImpl.h"
+
+namespace Microsoft {
+namespace Featurizer {
+namespace Featurizers {
+
+/////////////////////////////////////////////////////////////////////////
+///  \class         ForecastingPivotTransformer
+///  \brief         Expand output from upstream featurizers where all values
+///                 in the same column across matrixes are not null/empty
+///
+template <typename InputT>
+class ForecastingPivotTransformer :
+    public Components::InferenceOnlyTransformerImpl<
+        typename std::vector<Microsoft::Featurizer::RowMajMatrix<typename Microsoft::Featurizer::Traits<InputT>::nullable_type>>,
+        std::vector<typename Microsoft::Featurizer::Traits<InputT>::nullable_type>
+    > {
+public:
+    // ----------------------------------------------------------------------
+    // |
+    // |  Public Types
+    // |
+    // ----------------------------------------------------------------------
```

Comment:
>// ---------------------------------------------------------------------- [](start = 4, length = 73)

Add a static_assert here to communicate to the caller that they don't need to pass an optional type - it will be converted to optional for them:

`static_assert(Traits<InputT>::IsNullableType == false || Traits<InputT>::IsNativeNullableType, "'InputT' should not be a nullable type");`